iis3dwb_reg.c

```diff
@@ -435,11 +435,10 @@ int32_t iis3dwb_xl_axis_selection_set(stmdev_ctx_t *ctx,
 int32_t iis3dwb_xl_axis_selection_get(stmdev_ctx_t *ctx,
                                       iis3dwb_xl_axis_sel_t *val)
 {
-  iis3dwb_ctrl4_c_t ctrl4_c;
-  iis3dwb_ctrl6_c_t ctrl6_c;
-  int32_t ret;
+  iis3dwb_ctrl4_c_t ctrl4_c={0};
+  iis3dwb_ctrl6_c_t ctrl6_c={0};
 
-  ret = iis3dwb_read_reg(ctx, IIS3DWB_CTRL4_C, (uint8_t *)&ctrl4_c, 1);

```

Comment:
Well, I really prefer the first idea you had:

```
if (ret != 0) { return ret; }
```

Simpler and enough.
If ok, pls, apply everywhere.

Reply:
done.
You can check if now it's ok